src/helpers/userAnalyticHelpers.js

```diff
@@ -1,10 +1,41 @@
+import { EnvHelper } from "./Environment";
 import { retrieveUTMQueryParameters } from "./QueryParameterHelper";
 
+// Obtain country from IP address
+function countryLookup() {
+  // Determine the country the user is from, based on IP
+  // Geoapify offers 3000 lookups/day, so we should be fine
+  var apiKey = EnvHelper.getGeoapifyAPIKey();
+
+  fetch("https://api.geoapify.com/v1/ipinfo?apiKey=" + apiKey, {
+    method: "GET",
+  })
+    .then(function (response) {
+      if (response.ok) {
+        return response.json();
+      } else {
+        throw "Unable to determine country from IP lookup: " + response.body;
+      }
+    })
+    .then(function (json) {
+      return json.country.name;
+    })
+    .catch(function (error) {
+      console.log(error);
+      // Set the country to a default value
+      return "Unknown";
+    });
+}
+
 // Pushing data to segment analytics
 export function segmentUA(data) {
   const stringifiedData = JSON.stringify(data);
   var analytics = (window.analytics = window.analytics);
 
+  // We don't record the IP for privacy reasons, so we capture the country instead
+  var country = countryLookup();
+  data.country = country;
```

Comment:
This isn't going to work. You either need a callback in `countryLookup` or use the `async/await` pattern (recommended). 

Explanation:
the `fetch` method in `countryLookup` is a promise, which executes `.then()` once the promise resolves. The overall method doesn't wait for the promise to complete to return. 
In conditions where `countryLookup()` takes a while. `var country` is going to be undefined. 


YAY javascript.